haproxy_exporter.go

```diff
@@ -102,6 +102,8 @@ var (
 		16: newServerMetric("redispatch_warnings_total", "Total of redispatch warnings.", nil),
 		17: newServerMetric("up", "Current health status of the server (1 = UP, 0 = DOWN).", nil),
 		18: newServerMetric("weight", "Current weight of the server.", nil),
+		21: newServerMetric("check_fail", "Total number of failed health checks.", nil),
```

Comment:
Both of these values are counters, so we should append the `_total` suffix as described here https://prometheus.io/docs/practices/naming/#metric-names.

I have started to work on a fix of the metric types, so that these will also get exported correctly as counter and not gauge.